File: 1-two-sum/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    const map = {};

    for (let i = 0; i < nums.length; i++) {
        let goal = target - nums[i];
        if (goal in map) return [i, map[goal]];
        map[nums[i]] = i;
    }
    return;
};

File: 1-two-sum/two-sum.cs
public class Solution {
public int[] TwoSum(int[] nums, int target)
{
    IDictionary<int, int> map = new Dictionary<int, int>();

    int[] indices = new int[2];

    for (int i = 0; i < nums.Length; i++)
    {
        int goal = target - nums[i];
        if (map.ContainsKey(goal))
        {
            indices[0] = i;
            indices[1] = map[goal];
            return indices;
        }
        map.TryAdd(nums[i], i);
    }

    return indices;
}
}